pkg/driver/openstack-manila/openstack.go

```diff
@@ -0,0 +1,125 @@
+package openstack_manila
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/openstack"
+	"github.com/gophercloud/gophercloud/v2/openstack/sharedfilesystems/v2/sharetypes"
+	"github.com/gophercloud/utils/v2/openstack/clientconfig"
+	"github.com/openshift/csi-operator/pkg/openstack-manila/util"
+	"github.com/openshift/csi-operator/pkg/version"
+	"sigs.k8s.io/yaml"
+)
+
+type openStackClient struct {
+	cloud *clientconfig.Cloud
+}
+
+func NewOpenStackClient(cloudConfigFilename string) (*openStackClient, error) {
+	cloud, err := getCloudFromFile(cloudConfigFilename)
+	if err != nil {
+		return nil, err
+	}
+	return &openStackClient{
+		cloud: cloud,
+	}, nil
+}
+
+func (o *openStackClient) GetShareTypes() ([]sharetypes.ShareType, error) {
```

Comment:
I would like to rewrite all of this to just use `clouds.yaml` but that's a job for later...